Migrate verdi computer disable/enable to click

giovannipizzi opened the pull request:
This is a partial migration of verdi computer (see #1582).
Focuses on `enable` and `disable` only, tests added.

I extracted the work I was doing in a different branch to isolate only the changes to disable/enable, and allow other people to help with the commands. 

I will take care in a different PR of the interactive commands (setup and configure).

sphuber commented on aiida/backends/tests/cmdline/commands/test_computer.py:
```diff
@@ -0,0 +1,106 @@
+from aiida.common.exceptions import NotExistent
+
+
+# pylint: disable=missing-docstring
```
pylint disable does not seem necessary here? Even more so given that these files have not been added to pre-commit yet?

sphuber commented on aiida/backends/tests/cmdline/commands/test_computer.py:
```diff
@@ -0,0 +1,106 @@
+    """
+
+    @classmethod
+    def setUpClass(cls, *args, **kwargs):
```
the base `AiidaTestCase` already provides a `computer` class property. If this computer is just a base stud, maybe you could that one to avoid duplication

giovannipizzi commented:
ups... didn't realize there were already some tests for the old cli interface. I'm going to remove them.

sphuber commented on aiida/cmdline/commands/computer.py:
```diff
@@ -46,21 +49,17 @@ def prompt_for_computer_configuration(computer):
         except KeyError:
-            print >> sys.stderr, ("Internal error! "
-                                  "No {} getter defined in Computer".format(getter_name))
+            print >> sys.stderr, ("Internal error! " "No {} getter defined in Computer".format(getter_name))
```
print with `aiida.cmdline.utils.echo` based functions. the `echo_critical` will automatically print to stderr and exit

sphuber commented on aiida/cmdline/commands/computer.py:
```diff
@@ -46,21 +49,17 @@ def prompt_for_computer_configuration(computer):
         except KeyError:
-            print >> sys.stderr, ("Internal error! "
-                                  "No {} setter defined in Computer".format(setter_name))
+            print >> sys.stderr, ("Internal error! " "No {} setter defined in Computer".format(setter_name))
```
see above about echo

sphuber commented on aiida/cmdline/commands/computer.py:
```diff
@@ -123,6 +118,76 @@ def prompt_for_computer_configuration(computer):
+            echo.echo_info("Computer '{}' already disabled.".format(computer.name))
+        else:
+            computer.set_enabled_state(False)
+            echo.echo_info("Computer '{}' disabled.".format(computer.name))
```
Maybe for the cases where the (desired) change is actually effectuated, it would be good to use `echo_success` instead of `echo_info`?

giovannipizzi commented:
I temporarily added it to at least fix the formatting, but there were too many problems with prospector that will be anyway solved when finishing the conversion, so I disabled it for now.

giovannipizzi commented:
hi guys, do you have an idea why the tests are failing?
1. I fixed the code with yapf and locally it does not show anything to do, but it's still complaining on Travis
1. I get some mlocate problems, but I don't think I changed anything related to that, maybe something changed on the default Travis configuration?
